go/src/koding/klient/app/defertime.go

```diff
@@ -0,0 +1,82 @@
+package app
+
+import "time"
+
+// DeferTime allows to invoke registered function after provided duration. It
+// can cancel deferred function call even if internal timer was already started.
+type DeferTime struct {
```

Comment:
Sorry but i see it as another level of indirection. You moved code to an another callback and implemented the same functionality from time.Timer + better Stop functionality while still leaking go routines. 
Please check the code here https://github.com/koding/koding/pull/9561/files#diff-3b0ed4f4068baf2a0714d960aece64bdR59 there is no code block as return statement